src/EnergyPlus/Material.cc

```diff
@@ -2758,4 +2762,130 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
+                               "properties ({} Absorptance)",
+                               cCurrentModuleObject,
+                               addOnType,
+                               thisMaterial->Name,
+                               addOnType));
+        errorsFound = true;
+        return;
+    }
+
+    Material::VariableAbsCtrlSignal controlSignal = Material::VariableAbsCtrlSignal::SurfaceTemperature; // default value
+    controlSignal = static_cast<VariableAbsCtrlSignal>(getEnumerationValue(VariableAbsCtrlSignalUC, UtilityRoutines::MakeUPPERCase(alphas(3))));
+    int functionIdx = -1;
+    int scheduleIdx = -1;
+    if (controlSignal == VariableAbsCtrlSignal::Scheduled) {
+        if (alphas(5) == "") {
+            ShowSevereError(state, format("{}: Schedule not defined for object {}", cCurrentModuleObject, alphas(1)));
+            errorsFound = true;
+            return;
+        } else {
+            scheduleIdx = ScheduleManager::GetScheduleIndex(state, alphas(5));
+        }
+    } else { // controlled by performance table or curve
+        functionIdx = Curve::GetCurveIndex(state, alphas(4));
+        if (functionIdx == 0) {
+            ShowSevereError(state, format("Invalid {}={}", state.dataIPShortCut->cAlphaFieldNames(4), alphas(4)));
+            errorsFound = true;
+            return;
+        }
+    }
+
+    if (addOnType == "Thermal") {
+        thisMaterial->variableThermalAbsorptanceCtrlSignal = controlSignal;
+        thisMaterial->variableThermalAbsporptanceScheduleIdx = scheduleIdx;
+        thisMaterial->variableThermalAbsorptanceFunctionIdx = functionIdx;
+    } else if (addOnType == "Solar") {
+        thisMaterial->variableSolarAbsorptanceCtrlSignal = controlSignal;
+        thisMaterial->variableSolarAbsporptanceScheduleIdx = scheduleIdx;
+        thisMaterial->variableSolarAbsorptanceFunctionIdx = functionIdx;
+    }
+}
+
+void GetVariableThermalAbsorptanceInput(EnergyPlusData &state, bool &errorsFound)
+{
+    int numVariThermalAbs = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "MaterialProperty:VariableThermalAbsorptance");
+    for (int i = 1; i <= numVariThermalAbs; ++i) {
+        ReadingOneMaterialAddOn(state, i, "Thermal", errorsFound);
+    }
+}
+
+void GetVariableSolarAbsorptanceInput(EnergyPlusData &state, bool &errorsFound)
+{
+    int numVariSolarAbs = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "MaterialProperty:VariableSolarAbsorptance");
+    for (int i = 1; i <= numVariSolarAbs; ++i) {
+        ReadingOneMaterialAddOn(state, i, "Solar", errorsFound);
+    }
+}
+
+void GetMaterialAddOnInput(EnergyPlusData &state, bool &errorsFound)
+{
+    GetVariableThermalAbsorptanceInput(state, errorsFound);
+    GetVariableSolarAbsorptanceInput(state, errorsFound);
+}
+
+int GetMaterialNumFromSurfNum(EnergyPlusData &state, int surfNum)
```

Comment:
Move these functions into the `DataSurfaces` module. That way you won't have to include `Donstruction.hh` and `DataSurfaces.hh` here.